modules/navier_stokes/include/kernels/MDFluidKernelBase.h

```diff
@@ -0,0 +1,85 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Kernel.h"
+#include "SinglePhaseFluidProperties.h"
+#include "MooseVariable.h"
+
+/**
+ * This class couples together all the variables for the 3D fluid equations to allow them to be used
+ * in derived Kernel classes.
+ */
+class MDFluidKernelBase : public Kernel
+{
+public:
+  static InputParameters validParams();
+
+  MDFluidKernelBase(const InputParameters & parameters);
+  virtual ~MDFluidKernelBase() {}
+
+protected:
+  Real velocityDiv() { return _grad_u_vel[_qp](0) + _grad_v_vel[_qp](1) + _grad_w_vel[_qp](2); }
+  RealVectorValue velocityDot()
+  {
+    RealVectorValue vec_vel_dot(_u_vel_dot[_qp], _v_vel_dot[_qp], _w_vel_dot[_qp]);
+    return vec_vel_dot;
+  }
```

Comment:
In general I try to encourage people (including myself who's often a guilty party) to move multi-line definitions out of the class definition, e.g. we could make this inline in the header outside of the class definition